s390x assembly pack: fix minor issues

p-steuer opened the pull request:
Its been a while since PR #2859 . In the meantime I have reworked most of that code.
Id like to process all this in smaller requests, one at a time, as @dot-asm suggested.

So here comes the first part.

dot-asm commented:
I apologize that it dragged so long, I *will* get to it. And pushing for simpler things first is a way to engage :-) As for ghash. Just in case for other's reference, the problematic code sequence is actually if (0)-ed, which means that default build produces correct result. As for solution I suggest to solve it from the other way, see #4531. I did verify it on hardware, but please double-check it too.

dot-asm commented on crypto/aes/asm/aes-s390x.pl:
```diff
@@ -821,12 +821,9 @@ ()
 	llihh	%r0,0x8000
 	srlg	%r0,%r0,0(%r5)
+	ng	%r0,32(%r1)	# check km capability vector
 	ng	%r0,48(%r1)	# check kmc capability vector
```
Could you modify the commentary clarifying that we check for both km and kmc for given key length, and rely on the fact that bit mask is same [for basic operations]. Trouble here is is that it's not self-obvious to fresh eye...

dot-asm replied:
I meant modify commentary specifically here, in above two lines. ng-ing two values leaves first-time reader wondering what is that about, suspecting it's a copy-n-paste error. But if commentary said "we check availability of both km and kmc support for given key length", i.e. one commentary describing both checks, then it wouldn't like like copy-n-paste error.

p-steuer replied:
okay, done.

p-steuer commented:
@dot-asm : i verified #4531 , works for me. therefore i removed my patch. Modified the commentary to clarify that km and kmc use the same function code for a given key length. It follows that the bit masks are the same because the number of the corresponding bit equals the function code (in POP notation, ie bit 0 being the most-significant in big-endian).

Next PR is also a small one :-)  --> #4534

dot-asm commented:
Merged. Thanks!
